src/tracing/vizceral_json_file_creator_nodes.coffee

```diff
@@ -0,0 +1,179 @@
+redis = require('redis')
+_ = require 'underscore'
+conf = require '../tracing_conf'
+timeInterval = conf.query_interval
+###
+This script will get data from redis and prepare JSON from it which can be consumed by
+Vizceral to display pipeline data flow. This script will consider every topic as node
+@author manish@moz.com (Manish Ranjan)
+###
+
+module.exports = class VizJsonNode
+  # This method prepares the JSON and sends it as return as and when requested
+  @getJSON: (callback) ->
+    lastSavedTime = Date.now() - timeInterval
+    redis.createClient().zrangebyscore 'narrows-tracing', lastSavedTime, Date.now(), (err, allData) ->
+      return callback err if err
+
+      [success, errors] = filterDataAndError allData
+      topicToCount = getErrorMapOfTopicToCount errors
+      traceIdToTopicChannel = getTopicToChannelbyId success
+      reducedMap = reduceMap traceIdToTopicChannel
+      srcToTarget = getSourceToTarget reducedMap
+      entryNode = findEntryNodes srcToTarget
+      nodeList = getNodeList srcToTarget
+      srcToTargetList = getSourceToTargetList srcToTarget, entryNode, topicToCount
+      output = prepareJSON nodeList, srcToTargetList
+      callback null, output
+
+  # this function filters data and error data in two different lists based on the length of incoming data
+  filterDataAndError = (allData) ->
+    ###
+    Two kind of data 1. success 2. error
+    1.<Tsource:Csource|Tdestination| rev-id| Date.now() to keep it unique
+    2.<rev-id| Tdestination:Cdestination | Date.now()>
+    ###
+    errors = []
+    success = []
+    for key, value of allData
+      {length} = value.split '|'
+      if length is 3 then errors.push value else success.push value
+    [success, errors]
+
+  # this function gets the no of time error occured and creates a map based on that
+  getErrorMapOfTopicToCount = (errorData)->
+    errorNodeMap = {}
+    for key, value in errorData
+      node = key.split('|')[1].split(':')[0]
+      errorNodeMap[node] or= 0
+      errorNodeMap[node]++
+    errorNodeMap
+
+  # This function creates a  map of key(traceId=revNo + Date.Now()) to the topic channel communication
+  getTopicToChannelbyId = (success) ->
+    traceIdToTopicChannel = {}
+    #for value, index in success
+    for key, value in success
+      successData = key.split '|'
+      data = successData[0..1].join('>')  # Ts:Cs>Td
+      keyTrace = successData[2]
+      if keyTrace of traceIdToTopicChannel
+        traceIdToTopicChannel[keyTrace] = "#{traceIdToTopicChannel[keyTrace]},#{data}"
+      else
+        traceIdToTopicChannel[keyTrace] = data
+    traceIdToTopicChannel
```

Comment:
The function name and this variable name don't seem consistent about the data they are returning.